components/optimizer/__init__.py

```diff
@@ -0,0 +1,22 @@
+from .greedymax import GreedyMax
+from common.minimodel import Plans
+
+
+class Optimizer:
+    """
+    Entrypoint to interact with an BaseOptimizer object.
+    All algorithms need to follow the same structure to create a Plan
+    """
+
+    def __init__(self, selection, algorithm=GreedyMax(some_parameter=1)):
+        self.algorithm = algorithm.add(selection)
+    
+    def schedule(self, night_events) -> Plans:
+        # TODO: This forces to make Plans for each site SEPARATELY.
+        # This would create and issue with OR groups as observations can be schedule 
+        # on different sites but not twice. Old GM had some check-ins to handle this 
+        # cases but right now we don't have OR groups implemented.
```

Comment:
We should talk about this and see what we need to do. I had the impression that a NightlyPlan would be site-specific, but a collection of plans would include nightly plans for all sites. If this is going to be a design problem, though, we will have to do something else.